lib/service/connect.go

```diff
@@ -49,6 +50,8 @@ import (
 // reconnectToAuthService continuously attempts to reconnect to the auth
 // service until succeeds or process gets shut down
 func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*Connector, error) {
+	// TODO(fspmarshall): we should probably have a longer retry period for Instance certs
+	// in order to avoid catastrophic load in the event of an auth server downgrade.
```

Comment:
Auth downgrade without downgrading clients first is not a supported configuration tho 🤔

Reply:
Having this particular feature break with the norm and be back-combat with older servers was a deliberate deviation RJ and I decided to make.  Basically, we thought the potential negative fallout of heartbeats being totally non-functional if someone needed to revert their auth after a V10 upgrade was a bit too extreme.  While we don't officially support older auths, teleport usually "kinda works" in that scenario.  Cache might go into an error loop, and some specific features might be pretty buggy, but usually you could at least theoretically still ssh into nodes (even if w/ very poor perf).

This isn't unprecedented (e.g. a lot of the GRPC methods that used to be normal HTTP methods kept the old HTTP method as a fallback for at least one version).